src/Engine/EmitMSIL/Internals.cs

```diff
@@ -0,0 +1,233 @@
+        {
+            bool lhs = ToBoolean(a);
+            bool rhs = ToBoolean(b);
+
+            return lhs || rhs;
+        }
+
+        public static bool Not(dynamic a)
+        {
+            bool data = ToBoolean(a);
+
+            return !data;
+        }
+
+        public static bool eq(dynamic a, dynamic b)
+        {
+            if (a is bool || b is bool)
+            {
+                bool ab = ToBoolean(a);
+                bool bb = ToBoolean(b);
+                return ab == bb;
+            }
+
+            if (a is double || b is double || a is long || b is long)
+            {
+                if (a is double || b is double)
+                {
+                    double ad = a;
+                    double bd = b;
+                    return MathUtils.Equals(ad, bd);
+                }
+
+                return a == b;
+            }
+
+            if (a is string astr && b is string bstr)
+            {
+                return String.Compare(astr, bstr, StringComparison.InvariantCulture) == 0;
+            }
+
+            return a.Equals(b);
+        }
+
+        public static bool nq(dynamic a, dynamic b)
+        {
+            if (a is bool || b is bool)
+            {
+                bool ab = ToBoolean(a);
+                bool bb = ToBoolean(b);
+                return ab != bb;
+            }
+
+            if (a is double || b is double || a is long || b is long)
+            {
+                if (a is double || b is double)
+                {
+                    double ad = a;
+                    double bd = b;
+                    return !MathUtils.Equals(ad, bd);
+                }
+
+                return a != b;
+            }
+
+            if (a is string astr && b is string bstr)
+            {
+                return String.Compare(astr, bstr, StringComparison.InvariantCulture) != 0;
+            }
+
+            return !a.Equals(b);
+        }
+
+        public static bool gt(dynamic a, dynamic b)
+        {
+            if (a is double || b is double || a is long || b is long)
+            {
+                return a > b;
+            }
+
+            return false;
+        }
+
+        public static bool lt(dynamic a, dynamic b)
+        {
+            if (a is double || b is double || a is long || b is long)
+            {
+                return a < b;
+            }
+
+            return false;
+        }
+
+        public static bool ge(dynamic a, dynamic b)
+        {
+            if (a is double || b is double || a is long || b is long)
+            {
+                return a >= b;
+            }
+
+            return false;
+        }
+
+        public static bool le(dynamic a, dynamic b)
+        {
+            if (a is double || b is double || a is long || b is long)
+            {
+                return a <= b;
+            }
+
+            return false;
+        }
+
+        public static IEnumerable<MethodBase> GetInternalMethod(string name)
```

Comment:
This looks like it's always returning a single MethodBase object, no?

Reply:
For now yes.

Reply:
I was kind of preparing for overloads.